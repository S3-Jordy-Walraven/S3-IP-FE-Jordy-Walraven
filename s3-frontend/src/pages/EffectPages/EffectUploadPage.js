import React, { useRef, useState, useContext } from "react";
import "../../css/Form.css";
import Canvas from "../../components/EffectComponents/Canvas";
import { userContext } from "../../context/userContext";
import EffectService from "../../services/EffectService";
import { useNavigate } from "react-router-dom";

const EffectUploadPage = (props) => {
  const effectNameInput = useRef();
  const htmlInput = useRef();
  const [html, sethtml] = useState("");
  const [htmlString, sethtmlString] = useState("");
  const stateUser = useContext(userContext);
  const service = new EffectService();
  const navigate = useNavigate();

  async function submitHandler(event) {
    event.preventDefault();
    console.log(stateUser);
    let data = {
      effectName: effectNameInput.current.value,
      effectContent: htmlString,
      subjectId: stateUser.user.sub,
      creatorName: stateUser.user.name,
    };

    try {
      await service.createEffect(data);
      props.reloadCallback();
      navigate("/");
    } catch { }
  }

  function SetString(Html) {
    sethtmlString(Html);
  }

  return (
    <div data-testid="effectUploadPage-1">
      <Canvas
        fileInput={html}
        htmlStringCallback={SetString}
        finalString={htmlString}
      ></Canvas>
      <div className="formBody">
        <form className="form" onSubmit={submitHandler}>
          <div className="mb-3">
            <label className="form-label">
              EffectFile
              <input
                data-testid="htmlInput-1"
                className="form-control"
                type="file"
                accept=".html"
                id="EffectFile"
                ref={htmlInput}
                onChange={(e) => {
                  sethtml(document.getElementById("EffectFile").files[0]);
                }}
              />
            </label>
          </div>
          <div className="mb-3">
            <label className="form-label">
              Effect name
              <input
                type="Effect Name"
                className="form-control"
                id="effectName"
                ref={effectNameInput}
              />
            </label>
          </div>

          <button
            type="submit"
            className="btn btn-primary"
            data-testid="submitButton-1"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  );
};

export default EffectUploadPage;
