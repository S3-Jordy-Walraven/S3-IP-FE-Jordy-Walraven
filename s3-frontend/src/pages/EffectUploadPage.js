import React, { useRef, useState } from "react";
import "../css/Form.css";
import Canvas from "../components/Canvas";
import { userContext } from "../userContext";
import EffectService from "../services/EffectService";

const EffectUploadPage = () => {
  const effectNameInput = useRef();
  const htmlInput = useRef();
  const [html, sethtml] = useState("");
  const [htmlString, sethtmlString] = useState("");
  const [stateUser, setStateUser] = useState(null);
  const service = new EffectService();

  function submitHandler(event) {
    event.preventDefault();
    var data = {
      effectName: effectNameInput.current.value,
      effectContent: htmlString,
      subjectId: stateUser.user.sub,
      creatorName: stateUser.user.name,
    };
    console.log(data);
    service.createEffect(data);
  }

  function SetString(Html) {
    sethtmlString(Html);
  }

  return (
    <div>
      <userContext.Consumer>
        {(value) => setStateUser(value)}
      </userContext.Consumer>
      <Canvas fileInput={html} htmlStringCallback={SetString}></Canvas>
      <div className="formBody">
        <form className="form" onSubmit={submitHandler}>
          <div className="mb-3">
            <label className="form-label">
              EffectFile
              <input
                className="form-control"
                type="file"
                accept=".html"
                id="EffectFile"
                ref={htmlInput}
                onChange={(e) => {
                  const selectedFile = sethtml(
                    document.getElementById("EffectFile").files[0]
                  );
                }}
              />
            </label>
          </div>
          <div className="mb-3">
            <label className="form-label">
              Effect name
              <input
                type="Effect Name"
                className="form-control"
                id="effectName"
                ref={effectNameInput}
              />
            </label>
          </div>

          <button type="submit" className="btn btn-primary">
            Submit
          </button>
        </form>
      </div>
    </div>
  );
};

export default EffectUploadPage;
