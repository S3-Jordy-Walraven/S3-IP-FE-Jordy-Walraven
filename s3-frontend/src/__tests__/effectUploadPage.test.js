import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import EffectUploadPage from "../pages/EffectPages/EffectUploadPage"
import userEvent from "@testing-library/user-event"

afterEach(cleanup)

test("should render canvas", () => {

    render(<EffectUploadPage />);
    const effectUploadPageElement = screen.getByTestId("effectUploadPage-1");
    expect(effectUploadPageElement).toBeInTheDocument();
    expect(effectUploadPageElement).toHaveTextContent("EffectFile");
})

test("file upload", () => {
    render(
            <EffectUploadPage/>
    );

    const effectUploadPageElement = screen.getByTestId("effectUploadPage-1");
    const htmlInput = screen.getByTestId("htmlInput-1");
    userEvent.upload(htmlInput, new File(["<script>console.log('hello')</script>"], "test.html"));
    expect(effectUploadPageElement).toBeInTheDocument();
  
})

test("test on submit", () => {
    let isCalled = false;
    function ReloadEffects() {
        console.log("Reloading effects")
        isCalled = true;
    }

    render(
            <EffectUploadPage reloadCallback={ReloadEffects} />
    );

    const effectUploadPageElement = screen.getByTestId("effectUploadPage-1");
    const submitButton = screen.getByTestId("submitButton-1");
    userEvent.click(submitButton);
    expect(effectUploadPageElement).toBeInTheDocument();
    expect(isCalled).toBe(true);
})

test("matches snapshot", () => {

    const tree = renderer.create(<EffectUploadPage />).toJSON();
    expect(tree).toMatchSnapshot();
})