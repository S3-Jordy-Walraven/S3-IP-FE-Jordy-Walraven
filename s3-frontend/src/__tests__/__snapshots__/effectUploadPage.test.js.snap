// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="effectUploadPage-1"
>
  <div
    data-testid="canvas-1"
    id="wrap"
  >
    <iframe
      height={100}
      id="scaled-frame"
      sandbox="allow-scripts "
      srcDoc=""
      title="canvas"
      width="100%"
    />
  </div>
  <div
    className="formBody"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
        >
          EffectFile
          <input
            accept=".html"
            className="form-control"
            data-testid="htmlInput-1"
            id="EffectFile"
            onChange={[Function]}
            type="file"
          />
        </label>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
        >
          Effect name
          <input
            className="form-control"
            id="effectName"
            type="Effect Name"
          />
        </label>
      </div>
      <button
        className="btn btn-primary"
        data-testid="submitButton-1"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
